sklearn/linear_model/_ridge.py

```diff
@@ -1616,6 +1616,18 @@ def fit(self, X, y, sample_weight=None):
             self.alpha_ = gs.best_estimator_.alpha
             self.best_score_ = gs.best_score_
 
+        min_alpha = min(self.alphas)
+        max_alpha = max(self.alphas)
+        if self.alpha_ in [min_alpha, max_alpha]:
+            warnings.warn("The optimal value for the regularization parameter "
+                          "'alpha' was {}\n which lies at a boundary of the "
+                          "explored range (between {} and {}).\n Consider "
+                          "setting the 'alphas' parameter to explore a "
+                          "wider range.\n To ignore warnings try running "
+                          "\nimport warnings "
+                          "\nwarnings.filterwarnings('ignore')"
```

Comment:
I think we want this warning to be of a `StatisticalWarning`, and the user should either be bale to suppress this specific warning, or all `StatisticalWarnings`, not all of them.

Reply:
I you we can leave this instruction on how to ignore warnings out of the error message, but maybe we should put it in our FAQ...

Reply:
But I think it should be here. If we decide not to have the parameter to suppress the warning (which is what you prefer), I think it should be easy for the user to suppress them, and I find having to go to a link they don't know, or searching for it online is easy.